### DISCUSSION TITLE
Index of IO out of range (Set AO)

### DISCUSSION THREAD

**Discussion by FrancescoPolvi** [visose/Robots#206] [Q&A] [answered]
Hello!

I'm trying to set Analog Output values by using the "Set AO" component in Grasshopper, but I'm getting an error message in the Create Program component.
The error I get is the following: _"1. Solution exception: Index of IO is out of range. Parameter name: index"_

I'm using the latest version of Robots (1.6.2). I've tried the 1.5.1 and I was having the same issue. 

My values are in the domain 0.2 to 0.8, so they fit in the required domain -1 to 1.
I'm using the Analog Output number 0.

I'm sharing a screenshot of the last part of the code. To make it even more basic I used only the first item of the list of values.

Any idea of what the error could be?
Thank you very much in advance!

![image](https://user-images.githubusercontent.com/63741563/208151841-329f79a8-68c6-4db9-99e2-4b46ae195fce.png)

**Comment by visose** [comment, marked as answer]
Which robot are you using? does it have analog outputs configure in the XML file?

**Reply by FrancescoPolvi** [reply]
We're using a robot that was not present in the libraries. We found the 3dm but the xml wasn't made by me and I didn't check it! I added the analog outputs and everything is working well now. Thanks!
